Clients/Xamarin.Interactive.Client/IssueReport.cs

```diff
@@ -0,0 +1,138 @@
+﻿//
+// Author:
+//   Aaron Bockover <abock@xamarin.com>
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using CommonMark;
+using CommonMark.Formatters;
+using CommonMark.Syntax;
+
+using Xamarin.Interactive.Markdown;
+using Xamarin.Interactive.SystemInformation;
+
+namespace Xamarin.Interactive.Client
+{
+    sealed class IssueReport
+    {
+        readonly HostEnvironment host;
+
+        public IssueReport (HostEnvironment host)
+            => this.host = host ?? throw new ArgumentNullException (nameof (host));
+
+        public string GetIssueReportUrlForGitHub (bool includeReport = true)
+        {
+            var builder = new System.Text.StringBuilder ();
+            builder.Append ("https://github.com/Microsoft/workbooks/issues/new");
+            if (includeReport)
+                builder.Append ("?body=").Append (System.Net.WebUtility.UrlEncode (GetIssueReportForGitHub ()));
+            return builder.ToString ();
+        }
+
+        public string GetIssueReportForGitHub ()
+        {
+            var writer = new StringWriter ();
+            WriteIssueReportForGitHubAsync (writer).GetAwaiter ().GetResult ();
+            return writer.ToString ().TrimEnd ();
+        }
+
+        public Task WriteIssueReportForGitHubAsync (TextWriter writer)
```

Comment:
Why is this `async`? Nothing `async` happens in it.

Reply:
Because originally async stuff was happening (see `WriteEnvironmentMarkdownAsync`); not necessary any more.